packages/wonder-blocks-popover/components/close-button.js

```diff
@@ -34,26 +39,27 @@ export default class CloseButton extends React.Component<Props> {
     };
 
     render() {
-        const {light, onClose, "aria-label": ariaLabel} = this.props;
+        const {light, onClose, "aria-label": ariaLabel, style} = this.props;
         return (
-            <IconButton
-                icon={icons.dismiss}
-                aria-label={ariaLabel}
-                onClick={onClose}
-                kind={light ? "primary" : "tertiary"}
-                light={light}
-                style={styles.closeButton}
-            />
+            <PopoverContext.Consumer>
+                {({close}) => {
+                    if (close && onClose) {
+                        throw new Error(
+                            "You've specified 'onClose' on the content when using Popover. Please specify 'onClose' on the Popover instead",
+                        );
```

Comment:
Yes, there is a case when `onClose` is being passed down from `PopoverContent` and/or `PopoverContentCore`.
But now that you mention it, I think that it could be unnecessary to add the `onClose` prop to these components. At the end we are delegating the close behavior to the `Popover` component itself. 

Do you think is it better to remove this `onClose` prop from `PopoverContent` and/or `PopoverContentCore` and?

Reply:
If we can remove it we should.  It avoid what could be a common mistake completely.

Reply:
I'm removing it, I noticed that it reduces the API surface area: one prop less to deal with for PopoverContent, PopoverContentCore and CloseButton. Now, `onClose` is completely centralized in `Popover`.

Please correct me if I'm wrong... I think that we originally used this technique in `Modals` b/c we needed to cover both scenarios: Opening a modal using the `ModalLauncher` and opening a modal by itself (e.g. `StandardModal` or `OnePaneDialog`).

The difference with Popover is that the `PopoverDialog` depends on `Popover` to be opened, where as in Modals, `OnePaneDialog` can be independent from `ModalLauncher` in some situations.

Reply:
That sounds right to me.  Modals and popovers are similar but there's some differences too.  Thanks for removing the prop.